src/components/deploymentConfig/DeploymentTemplateView.tsx

```diff
@@ -0,0 +1,831 @@
+                        value={selectedChart}
+                        classNamePrefix="chart_select"
+                        isOptionDisabled={(option) => !option.id}
+                        isSearchable={false}
+                        components={{
+                            IndicatorSeparator: null,
+                            DropdownIndicator,
+                            Option,
+                            MenuList: ChartMenuList,
+                        }}
+                        styles={{
+                            control: (base, state) => ({
+                                ...base,
+                                minHeight: '32px',
+                                boxShadow: 'none',
+                                border: 'none',
+                                cursor: 'pointer',
+                            }),
+                            valueContainer: (base, state) => ({
+                                ...base,
+                                padding: '0',
+                                fontWeight: '600',
+                            }),
+                            option: (base, state) => {
+                                return {
+                                    ...base,
+                                    color: 'var(--N900)',
+                                    backgroundColor: state.isFocused ? 'var(--N100)' : 'white',
+                                }
+                            },
+                            container: (base, state) => {
+                                return {
+                                    ...base,
+                                    width: '100%',
+                                }
+                            },
+                            menu: (base, state) => ({
+                                ...base,
+                                margin: '0',
+                                width: '250px',
+                            }),
+                            menuList: (base) => {
+                                return {
+                                    ...base,
+                                    position: 'relative',
+                                    paddingBottom: '0px',
+                                    maxHeight: '250px',
+                                }
+                            },
+                            dropdownIndicator: (base, state) => ({
+                                ...base,
+                                color: 'var(--N400)',
+                                padding: '0 8px',
+                                transition: 'all .2s ease',
+                                transform: state.selectProps.menuIsOpen ? 'rotate(180deg)' : 'rotate(0deg)',
+                            }),
+                        }}
+                        onChange={(selected) => {
+                            let filteredCharts = charts.filter((chart) => chart.name == selected.name)
```

Comment:
extract this method to separate function

Reply:
Done